docs/edgeql/operators/json.rst

```diff
@@ -39,17 +39,51 @@ JSON values can also be cast back into scalars. This casting is
 symmetrical meaning that if a scalar can be cast into JSON, only that
 particular JSON type can be cast back into that scalar:
 
-- JSON *string* can be cast into :eql:type:`str`
+- JSON *string* can be cast into :eql:type:`str`. Casting
+  :eql:type:`uuid` and :ref:`date and time types
+  <ref_datamodel_scalars_datetime>` to JSON results in a JSON
+  *string* representing the original value. This means that it is
+  also possible to cast a JSON *string* back into these types. The
+  string value has to be properly formatted (much like in case of
+  a :eql:type:`str` value being cast) or else the cast will raise an
+  exception.
 - JSON *number* can be cast into any of
   the :ref:`numeric types <ref_datamodel_scalars_numeric>`
 - JSON *boolean* can be cast into :eql:type:`bool`
 - JSON *null* is special since it can be cast into an ``{}`` of any type
 - JSON *array* can be cast into any valid EdgeDB array, so it must be
   homogeneous, and must not contain *null*
-- JSON *object* can be cast into a valid EdgeDB named tuple, but it must not
-  contain *null*
 
-The contents of JSON *objects* and *arrays* can also be accessed via ``[]``:
+*Unnamed* :eql:type:`tuple` is converted into a JSON *array* when cast
+into :eql:type:`json`. Whereas *named* :eql:type:`tuple` is converted
+into a JSON *object*. These casts are not reversible, i.e. it is not
+possible to cast a JSON value directly into a :eql:type:`tuple`.
```

Comment:
I would omit the last sentence, as this is not a fundamental limitation but a TODO.  Or add a note to the json casting issue to remember to update this bit once it works

Reply:
I'd rather add the note to the casting issue. Which one? #251?